paysage/backends/pytorch_backend/matrix.py

```diff
@@ -1378,4 +1378,33 @@ def fast_energy_distance(minibatch: T.FloatTensor,
             d3 += euclidean_distance(X[i], Y[j])
     d3 = d3 / (n*m)
 
-    return 2.0 * d3 - d2 - d1
+    return float_tensor(numpy.array([2.0 * d3 - d2 - d1]))
+
+
+def int_tensor(tensor: T.Tensor) -> T.FloatTensor:
+    """
+    Cast tensor to a float tensor.
+
+    Args:
+        tensor: A tensor.
+
+    Returns:
+        tensor: Tensor converted to floating point.
+
+    """
+    try:
+        # tensor is a numpy object
+        return torch.IntTensor(tensor.astype(int))
+    except Exception:
+        # tensor is a torch object
+        return tensor.int()
+
+
+def onehot(tensor, n_classes: int) -> T.FloatTensor:
```

Comment:
Unfortunately, the sparse tensors do not support dot products and scatter_ does not support tensors of different shapes, i.e. the output tensor is of shape (n_samples, n_classes) while the input tensor has shape (n_samples,). So for now this should be OK, I guess.